libraries/AP_RangeFinder/AP_RangeFinder_VL53L1X.cpp

```diff
@@ -0,0 +1,460 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  driver for ST VL53L1X lidar
+
+  Many thanks to Pololu, https://github.com/pololu/vl53l1x-arduino and
+  the ST example code
+ */
+#include "AP_RangeFinder_VL53L1X.h"
+
+#include <utility>
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/utility/sparse-endian.h>
+#include <stdio.h>
+
+extern const AP_HAL::HAL& hal;
+
+/*
+   The constructor also initializes the rangefinder. Note that this
+   constructor is not called until detect() returns true, so we
+   already know that we should setup the rangefinder
+*/
+AP_RangeFinder_VL53L1X::AP_RangeFinder_VL53L1X(RangeFinder::RangeFinder_State &_state, AP_HAL::OwnPtr<AP_HAL::I2CDevice> _dev)
+    : AP_RangeFinder_Backend(_state)
+    , dev(std::move(_dev)) {}
+
+/*
+   detect if a VL53L1X rangefinder is connected. We'll detect by
+   trying to take a reading on I2C. If we get a result the sensor is
+   there.
+*/
+AP_RangeFinder_Backend *AP_RangeFinder_VL53L1X::detect(RangeFinder::RangeFinder_State &_state, AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev)
+{
+    if(!dev) {
```

Comment:
Still not correct, you need a space after the if. Check our style guide: http://ardupilot.org/dev/docs/style-guide.html

Reply:
Hi @OXINARF,

I'm truly sorry for the mistakes! 
I have fixed them again, but I also noticed similar mistakes in VL53L0X library. Should I go ahead and fix them also?

Reply:
Great, no need to fix the other driver.  We can handle that as a separate PR.